File: Day31/CPP/reverse_Array_2Pointer.cpp
#include<bits/stdc++.h>
using namespace std;
void reverse(int i,vector<int> &nums,int n){
    if(i>=n) return;
    swap(nums[i],nums[n]);
    reverse(i+1, nums,n-1);
}
vector<int> reverseArray(int n, vector<int> &nums)
{
    reverse(0,nums,n-1);
    return nums;
}
int main()
{
    vector<int> nums = {10,20,30,40,50,60};
    vector<int> ans = reverseArray(nums.size(),nums);
    for(auto it:ans){
        cout<<it<<" ";
    }
    return 0;
}

File: Day31/Java/reverse_Array_2Pointer.java
public class reverse_Array_2Pointer {
    static void swap(int i,int n,int[] arr){
        int temp = arr[i];
        arr[i] = arr[n];
        arr[n] = temp;
    }
    static void reverse(int i,int[] nums,int n){
        if(i>=n) return;
        swap(i,n,nums);
        reverse(i+1,nums,n-1);
    }
    public int[] reverseArray(int n, int[] nums) {
        reverse(0,nums,n-1);
        return nums;
    }
    public static void main(String[] args) {
        int[] arr = {1,5,3,2,4};
        reverse_Array_2Pointer obj = new reverse_Array_2Pointer();
        int[] ans = obj.reverseArray(arr.length,arr);
        for(int i:ans){
            System.out.print(i+" ");
        }
    }
}